check_copyright.py

```diff
@@ -303,9 +307,11 @@ def has_valid_copyright(file_name: str, mime: str, is_on_ignore: bool, config_se
                 if mime == MIME['python']:
                     template = '# SPDX-FileContributor: ' + config_section['espressif_copyright']
                 candidate_line = template.format(years=format_years(years[0], file_name))
-                no_time_update = template.format(years=format_years(years[0], file_name, years[1]))
-                if code_lines[comment.line_number() - 1] != no_time_update:
-                    # update the line only in cases when not only the dates are changing
+                no_time_update = template.format(years=format_years(years[0], file_name, years[1] or years[0]))
```

Comment:
It would be nice to avoid duplications, but it will require a more significant refactor, so I guess, it's out of the scope of this change

Reply:
@kumekay agreed, the two checks for `SPDX-FileCopyrightText`  and  `SPDX-FileContributor`  seems identical, except for the `SPDX-` part, which could be parameterized in some common function.

Reply:
@kumekay Thank you very much for your review! I will use the longer names as you suggested. As for the other comments(tests, code duplication) I think it would probably make sense to manage these in separate PRs.